File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    # 각 수포자의 답안 패턴 정의
    patterns = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    # 각 수포자의 정답 수를 저장할 리스트
    scores = [0, 0, 0]
    
    # 정답과 비교하여 점수 계산
    for i in range(len(answers)):
        for j in range(3):
            if answers[i] == patterns[j][i % len(patterns[j])]:
                scores[j] += 1
    
    # 가장 많은 문제를 맞힌 사람 찾기
    max_score = max(scores)
    result = [i + 1 for i, score in enumerate(scores) if score == max_score]
    
    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> solution(int[] answers) {
        List<Integer> answer = new ArrayList<>();
        int[] arr1 = {1,2,3,4,5};
        int[] arr2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] arr3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[3];
        
        for(int i =0;i<answers.length;i++){
            if(answers[i]==arr1[i%5]) score[0]++;
            if(answers[i]==arr2[i%8]) score[1]++;
            if(answers[i]==arr3[i%10]) score[2]++;
        }
        
        int maxScore=Math.max(score[0],Math.max(score[1],score[2]));
        
        for(int i=0;i<3;i++){
            if(maxScore==score[i]) answer.add(i+1);
        }

        return answer;
    }
}